internal/controller/nnf_storage_controller.go

```diff
@@ -454,41 +457,85 @@ func (r *NnfStorageReconciler) aggregateNodeBlockStorageStatus(ctx context.Conte
 func (r *NnfStorageReconciler) createNodeStorage(ctx context.Context, storage *nnfv1alpha1.NnfStorage, allocationSetIndex int) (*ctrl.Result, error) {
 	log := r.Log.WithValues("NnfStorage", types.NamespacedName{Name: storage.Name, Namespace: storage.Namespace})
 
-	mgsAddress := storage.Spec.AllocationSets[allocationSetIndex].MgsAddress
 	if storage.Spec.FileSystemType == "lustre" {
+		mgsAddress := storage.Spec.AllocationSets[allocationSetIndex].MgsAddress
+
 		mgsNode := ""
-		for i, allocationSet := range storage.Spec.AllocationSets {
-			if allocationSet.TargetType == "mgt" || allocationSet.TargetType == "mgtmdt" {
-				// Wait for the MGT to be set up before creating nnfnodestorages for the other allocation sets
-				if allocationSetIndex != i {
-					if storage.Status.AllocationSets[i].Ready == false {
-						return nil, nil
+		if mgsAddress == "" {
+			for i, allocationSet := range storage.Spec.AllocationSets {
+				if allocationSet.TargetType == "mgt" || allocationSet.TargetType == "mgtmdt" {
+					// Wait for the MGT to be set up before creating nnfnodestorages for the other allocation sets
+					if allocationSetIndex != i {
+						if storage.Status.AllocationSets[i].Ready == false {
+							return nil, nil
+						}
+
 					}
+
+					mgsNode = allocationSet.Nodes[0].Name
+
+				}
+			}
+
+			if mgsNode != "" {
+				nnfNode := &nnfv1alpha1.NnfNode{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "nnf-nlc",
+						Namespace: mgsNode,
+					},
 				}
 
-				mgsNode = allocationSet.Nodes[0].Name
+				if err := r.Get(ctx, client.ObjectKeyFromObject(nnfNode), nnfNode); err != nil {
+					return &ctrl.Result{}, dwsv1alpha2.NewResourceError("could not get NnfNode: %v", client.ObjectKeyFromObject(nnfNode)).WithError(err)
+				}
+
+				mgsAddress = nnfNode.Status.LNetNid
 			}
 		}
 
-		if mgsNode != "" {
-			nnfNode := &nnfv1alpha1.NnfNode{
+		// Save the MGS address in the status section so we don't have to look in the NnfNodeStorage
+		storage.Status.MgsAddress = mgsAddress
+
+		// Create the NnfLustreMGT resource if this allocation set is for an MGT
+		allocationSet := storage.Spec.AllocationSets[allocationSetIndex]
+		if allocationSet.TargetType == "mgt" || allocationSet.TargetType == "mgtmdt" {
+			nnfLustreMgt := &nnfv1alpha1.NnfLustreMGT{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      "nnf-nlc",
+					Name:      storage.GetName(),
 					Namespace: mgsNode,
 				},
+				Spec: nnfv1alpha1.NnfLustreMGTSpec{
+					Addresses:   []string{mgsAddress},
+					FsNameStart: "aaaaaaaa",
```

Comment:
Why `aaaaaaaa`?

Reply:
The fsnames just increment (e.g., `aaaaaaaa`, `aaaaaaab`, `aaaaaaac`). Starting at `aaaaaaaa` was an arbitrary choice since it's easy to recognize that it's the start of the sequence and that `zzzzzzzz` is the end. Do you have another suggestion?

Reply:
Nope. Makes sense.